pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -113,28 +117,79 @@ export function readDeploymentProfiles(): settings.DeploymentProfileType {
   return profiles;
 }
 
+/**
+ * Read and parse plutil deployment profile files.
+ * @param rootPath the system or user directory containing profiles.
+ * @param defaultsPath the file path to the 'defaults' file.
+ * @param lockedPath the file path to the 'locked' file.
+ * @returns the defaults and/or locked objects if they exist, or
+ *          throws an exception if there is an error parsing the locked file.
+ */
+async function parseJsonFromPlist(rootPath: string, defaultsPath: string, lockedPath: string) {
+  let defaults;
+  let locked;
+
+  const plutilArgArray = ['-convert', 'json', '-r', '-o', '-', '--'];
+  let plutilPath = join(rootPath, defaultsPath);
+  let plutilArgs = plutilArgArray.concat(plutilPath);
+
+  if (settings.fileExists(plutilPath)) {
+    try {
```

Comment:
You could remove the duplication by changing this function to `parsePlistFiles(rootPath: string, defaultsPath: string, lockedPath: string)` and then create a helper function
```
async function parseJsonFromPlist(path: string) : RecursivePartial<Settings>|undefined {
if (!settings.fileExists(plutilPath)) {
  return;
}
const args = ['-convert', 'json', '-r', '-o', '-', '--', path];
try {
  const result = await spawnFile('plutil', args, { stdio: 'pipe' });
  try {
    return JSON.parse(result.stdout);
  } catch (error: any) {
    console.log(`Error parsing deployment profile JSON object for plist ${ path }\n${ error }`);
    throw error;
  }
} catch(error: any) {
  console.log(`Error parsing deployment profile ${ path }\n${ error }`);
  throw error;
}
} 

async function parseJsonFromPlist(rootPath: string, defaultsPath: string, lockedPath: string) {
  const locked = await parseJsonFromPlist(join(rootPath, defaultsPath));
  const default = await parseJsonFromPlist(join(rootPath, lockedPath));
  return { defaults, locked };
}

Reply:
I get four failing unit tests when I run this in `pkg/rancher-desktop/config/__tests__/settings.spec.ts` at lines 287 -- all of the failures are whre all fields should be locked, but it reports `undefined`. 

Not sure how the checks passed, but these need to pass locally as well.